doc/design/multiprocess.md

```diff
@@ -1,72 +1,264 @@
-return values are automatically serialized and sent between processes. Object
-references and `std::function` arguments are automatically tracked and mapped
-to allow invoked code to call back into invoking code at any time, and there is
-a 1:1 threading model where any thread invoking a method in another process has
-a corresponding thread in the invoked process responsible for executing all
-method calls from the source thread, without blocking I/O or holding up another
-call, and using the same thread local variables, locks, and callbacks between
-calls. The forwarding, tracking, and threading is implemented inside the
-[libmultiprocess](https://github.com/chaincodelabs/libmultiprocess) library
-which has the design goal of making calls between processes look like calls in
-the same process to the extent possible.
+</td></tr><tr><td>
+<code>Chain::getBlockHash</code> call diagram
+</td></tr></table>
+
+1. **Initiation in bitcoin-wallet**
+   - The wallet process calls the `getBlockHash` method on a `Chain` object. This method is defined as a virtual method in [`src/interfaces/chain.h`](../../src/interfaces/chain.h).
+
+2. **Translation to Cap’n Proto RPC**
+   - The `Chain::getBlockHash` virtual method is overridden by the `Chain` [client subclass](#c-client-subclasses-in-generated-code) to translate the method call into a Cap’n Proto RPC call.
+   - The client subclass is automatically generated by the `mpgen` tool from the [`chain.capnp`](https://github.com/ryanofsky/bitcoin/blob/pr/ipc/src/ipc/capnp/chain.capnp) file in [`src/ipc/capnp/`](../../src/ipc/capnp/).
+
+3. **Request Preparation and Dispatch**
+   - The `getBlockHash` method of the generated `Chain` client subclass in `bitcoin-wallet` populates a Cap’n Proto request with the `height` parameter, sends it to `bitcoin-node` process, and waits for a response.
+
+4. **Handling in bitcoin-node**
+   - Upon receiving the request, the Cap'n Proto dispatching code in the `bitcoin-node` process calls the `getBlockHash` method of the `Chain` [server class](#c-server-classes-in-generated-code).
+   - The server class is automatically generated by the `mpgen` tool from the [`chain.capnp`](https://github.com/ryanofsky/bitcoin/blob/pr/ipc/src/ipc/capnp/chain.capnp) file in [`src/ipc/capnp/`](../../src/ipc/capnp/).
+   - The `getBlockHash` method of the generated `Chain` server subclass in `bitcoin-wallet` receives a Cap’n Proto request object with the `height` parameter, and calls the `getBlockHash` method on its local `Chain` object with the provided `height`.
+   - When the call returns, it encapsulates the return value in a Cap’n Proto response, which it sends back to the `bitcoin-wallet` process,
+
+5. **Response and Return**
+   - The `getBlockHash` method of the generated `Chain` client subclass in `bitcoin-wallet` which sent the request now receives the response.
+   - It extracts the block hash value from the response, and returns it to the original caller.
+
+<table><tr><td>
```

Comment:
```suggestion
</td></tr></table>
```

I think you meant to close/end the table here.

Reply:
> I think you meant to close/end the table here. Otherwise this looks a bit weird rendered on GitHub.

Good catch should be fixed now